### ISSUE TITLE
Tab-completion in ex mode not working after update to 200.8

### ISSUE THREAD

**Issue by b-rodrigues** [syl20bnr/spacemacs#8630]
Hi
after updating to 200.8, it looks like tab-completion does not work when I go into ex mode: for example `:cl<tab>` to get :close, I get the following message «Wrong number of arguments (2 . 2), 3».

I've updated all of my packages too, and there are no warnings nor error messages when I start spacemacs.

**Comment by twillis** [comment]
experiencing the same error in dired mode trying to do anything (move/copy/rename etc...)

**Comment by rgrinberg** [comment]
I get this error when I type a space in any kind of helm query.

**Comment by aruiz14** [comment]
+1, also when using `:vsp fil<Tab>`

**Comment by dubnde** [comment]
+1 also using `SPC p f` and having  space in helm-query

**Comment by Apakollaps** [comment]
As @rgrinberg says it seems to affect all helm queries, backtrace indicates that it happens in `helm-fuzzy-matching-default-sort-fn`.

**Comment by bmag** [comment]
Can anyone post a backtrace? I'll look into this in several hours.

**Comment by dubnde** [comment]
After running `SPC p f` and a space after some helm-query

```
Debugger entered--Lisp error: (wrong-type-argument window-live-p nil)
  #<subr select-window>(nil norecord)
  ad-Advice-select-window(#<subr select-window> nil norecord)
  apply(ad-Advice-select-window #<subr select-window> (nil norecord))
  select-window(nil norecord)
  helm--autoresize-hook()
  run-hooks(helm-after-update-hook)
  helm-log-run-hook(helm-after-update-hook)
  #[0 "\301\206\303 \304\236\204\305!\210\306\307!\210\210\300\203#\310\311\300\"\210\312\300\301\"\210\313\211\207" [nil nil helm--force-updating-p helm-get-current-source candidates-process helm-display-mode-line helm-log-run-hook helm-after-update-hook helm-log "Update preselect candidate %s" helm-preselect nil] 3]()
  helm-update()
  helm-check-new-input(#("la " 0 3 (ws-butler-chg chg)))
  helm-check-minibuffer-input()
  #[0 "\206	?\205 \302 r\303\304\305\306\307!\310\"\311$\216\312 \210\313 *\262\207" [helm-in-persistent-action helm-suspend-update-flag internal--before-save-selected-window make-byte-code 0 "\301\300!\207" vconcat vector [internal--after-save-selected-window] 2 helm-check-minibuffer-input helm-print-error-messages] 7]()
  apply(#[0 "\206	?\205 \302 r\303\304\305\306\307!\310\"\311$\216\312 \210\313 *\262\207" [helm-in-persistent-action helm-suspend-update-flag internal--before-save-selected-window make-byte-code 0 "\301\300!\207" vconcat vector [internal--after-save-selected-window] 2 helm-check-minibuffer-input helm-print-error-messages] 7] nil)
  timer-event-handler([t 0 0 10000 repeat #[0 "\206	?\205 \302 r\303\304\305\306\307!\310\"\311$\216\312 \210\313 *\262\207" [helm-in-persistent-action helm-suspend-update-flag internal--before-save-selected-window make-byte-code 0 "\301\300!\207" vconcat vector [internal--after-save-selected-window] 2 helm-check-minibuffer-input helm-print-error-messages] 7] nil idle 0])
  read-from-minibuffer("[.spacemacs.d] Find file: " nil (keymap (tab . helm-execute-persistent-action) (33554440 . describe-key) (17 . ace-jump-helm-line) (f13 lambda nil (interactive) (helm-select-nth-action 12)) (f12 lambda nil (interactive) (helm-select-nth-action 11)) (f11 lambda nil (interactive) (helm-select-nth-action 10)) (f10 lambda nil (interactive) (helm-select-nth-action 9)) (f9 lambda nil (interactive) (helm-select-nth-action 8)) (f8 lambda nil (interactive) (helm-select-nth-action 7)) (f7 lambda nil (interactive) (helm-select-nth-action 6)) (f6 lambda nil (interactive) (helm-select-nth-action 5)) (f5 lambda nil (interactive) (helm-select-nth-action 4)) (f4 lambda nil (interactive) (helm-select-nth-action 3)) (f3 lambda nil (interactive) (helm-select-nth-action 2)) (f2 lambda nil (interactive) (helm-select-nth-action 1)) (menu-bar keymap (help-menu keymap (describe keymap (describe-mode . helm-help)))) (help keymap (109 . helm-help)) (23 . #[0 "\306\307\310\311\312\305!\313\"\314$\216\315\302!\210\316\304!\317\211\211\301=\203'\315\302!\210\320\202]\321\303\"\211\262\203:\315A!\210\320\202]\312!\262\322\300\"\262\323!\203S\315!\210\202\\\324\325\"\244\317\266\204\205e\202)\207" [#0 23 helm-yank-text-at-point ((31 . helm-undo-yank-text-at-point)) nil nil make-byte-code 0 "\300\205\300 \207" vconcat vector [] 1 call-interactively read-key nil t assoc lookup-key commandp mapcar identity last-command-event unread-command-events] 7 nil nil]) (f1 lambda nil (interactive) (helm-select-nth-action 0)) (8 . helm-next-source) (20 . helm-toggle-resplit-and-swap-windows) (C-tab . undefined) (67108897 . helm-toggle-suspend-update) (3 keymap (57 lambda nil (interactive) (helm-execute-selection-action-at-nth 9)) (56 lambda nil (interactive) (helm-execute-selection-action-at-nth 8)) (55 lambda nil (interactive) (helm-execute-selection-action-at-nth 7)) (54 lambda nil (interactive) (helm-execute-selection-action-at-nth 6)) (53 lambda nil (interactive) (helm-execute-selection-action-at-nth 5)) (52 lambda nil (interactive) (helm-execute-selection-action-at-nth 4)) (51 lambda nil (interactive) (helm-execute-selection-action-at-nth 3)) (50 lambda nil (interactive) (helm-execute-selection-action-at-nth 2)) (49 lambda nil (interactive) (helm-execute-selection-action-at-nth 1)) (63 . helm-help) (62 . helm-toggle-truncate-line) (21 . helm-refresh) (6 . helm-follow-mode) (9 . helm-copy-to-buffer) (11 . helm-kill-selection-and-quit) (25 . helm-yank-selection) (45 . helm-swap-windows)) (67108987 . helm-enlarge-window) (67108989 . helm-narrow-window) (19 . undefined) (24 keymap (57 lambda nil (interactive) (helm-execute-selection-action-at-nth -9)) (56 lambda nil (interactive) (helm-execute-selection-action-at-nth -8)) (55 lambda nil (interactive) (helm-execute-selection-action-at-nth -7)) (54 lambda nil (interactive) (helm-execute-selection-action-at-nth -6)) (53 lambda nil (interactive) (helm-execute-selection-action-at-nth -5)) (52 lambda nil (interactive) (helm-execute-selection-action-at-nth -4)) (51 lambda nil (interactive) (helm-execute-selection-action-at-nth -3)) (50 lambda nil (interactive) (helm-execute-selection-action-at-nth -2)) (49 lambda nil (interactive) (helm-execute-selection-action-at-nth -1)) (2 . helm-resume-list-buffers-after-quit) (98 . helm-resume-previous-session-after-quit) (6 . helm-quit-and-find-file)) (11 . helm-previous-line) (67108896 . helm-toggle-visible-mark) (0 . helm-toggle-visible-mark) (C-M-up . helm-scroll-other-window-down) (C-M-down . helm-scroll-other-window) (M-prior . helm-scroll-other-window-down) (M-next . helm-scroll-other-window) (12 . "
") (15 . helm-next-source) (10 . helm-next-line) (26 . helm-select-action) (9 . helm-execute-persistent-action) (13 . helm-maybe-exit-minibuffer) (left . helm-previous-source) (right . helm-next-source) (7 . helm-keyboard-quit) (22 . helm-next-page) (27 keymap (8388640 . spacemacs/helm-navigation-transient-state/body) (110 . next-history-element) (112 . previous-history-element) (115 . undefined) (5 . helm-display-all-sources) (1 . helm-show-all-in-this-source-only) (85 . helm-unmark-all) (97 . helm-mark-all) (109 . helm-toggle-all-marks) (41 . helm-next-visible-mark) (40 . helm-prev-visible-mark) (91) (32 . spacemacs/helm-navigation-transient-state/body) (33554454 . helm-scroll-other-window-down) (25 . helm-scroll-other-window-down) (22 . helm-scroll-other-window) (12 . helm-reposition-window-other-window) (111 . helm-previous-source) (62 . helm-end-of-buffer) (60 . helm-beginning-of-buffer) (118 . helm-previous-page)) (next . helm-next-page) (prior . helm-previous-page) (C-up . helm-follow-action-backward) ...) nil nil #("nil" 0 3 (fontified t)) t)
  helm-read-pattern-maybe("[.spacemacs.d] Find file: " nil nil nil nil nil nil)
  helm-internal((helm-source-projectile-dired-files-list helm-source-projectile-files-list) nil "[.spacemacs.d] Find file: " nil nil "*helm projectile*" nil nil nil)
  apply(helm-internal ((helm-source-projectile-dired-files-list helm-source-projectile-files-list) nil "[.spacemacs.d] Find file: " nil nil "*helm projectile*" nil nil nil))
  helm((helm-source-projectile-dired-files-list helm-source-projectile-files-list) nil "[.spacemacs.d] Find file: " nil nil "*helm projectile*" nil nil nil)
  apply(helm ((helm-source-projectile-dired-files-list helm-source-projectile-files-list) nil "[.spacemacs.d] Find file: " nil nil "*helm projectile*" nil nil nil))
  helm(:sources (helm-source-projectile-dired-files-list helm-source-projectile-files-list) :buffer "*helm projectile*" :truncate-lines nil :prompt "[.spacemacs.d] Find file: ")
  helm-projectile-find-file(nil)
  funcall-interactively(helm-projectile-find-file nil)
  call-interactively(helm-projectile-find-file nil nil)
  command-execute(helm-projectile-find-file)

```

**Comment by bmag** [comment]
@dubnde thanks, I see you have a different error than other people.

Can anyone with the wrong-number-of-arguments error also post a backtrace?

**Comment by dubnde** [comment]
oh I actually have the same. For some reason when I try to copy it it changes. Let me try again. My *backtrace* buffer disapears for some reason.

**Comment by twillis** [comment]
@bmag Sorry for the screenshot, the backtrace buffer kept disappearing whenI tried to focus to it.
![screen shot 2017-04-04 at 7 24 00 am](https://cloud.githubusercontent.com/assets/183699/24654501/e130553a-1907-11e7-8ac6-82872945bb34.png)

**Comment by alphapapa** [comment]
That change to `helm-fuzzy-matching-default-sort-fn` caused it, but here it's being called by `helm-flx-fuzzy-matching-sort`, so I guess it should be fixed in `helm-flx`, since the removed argument was obsolete anyway.

**Comment by bmag** [comment]
As @alphapapa says, this is an upstream issue in `helm-flx` that will hopefully be fixed soon (there's already a PR by @alphapapa, thank you). Reported upstream [here](https://github.com/PythonNut/helm-flx/issues/25) (thanks @twillis).

Once the issue is fixed upstream and the new version is available on MELPA, just update your packages. For now, I suggest everyone to add `helm-flx` to `dotspacemacs-excluded-packages`.

**Comment by dubnde** [comment]
FYI now fixed upstream. I update packages and don't see issue anymore.
